### PR TITLE
4.8.15 dev v4

### PR DESCRIPTION
**Pull request by anilkumarkammalapalli**

### CODE REVIEW

File: src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java [hunk 1]
```diff
@@ -432,4 +432,153 @@ private String validateSubSectorCreateReq(Map<String, Object> sectorRequest) {
...
+			return errMsg;
+		}
+		List<String> missingParams = new ArrayList<>();
+		String name = (String) designationRequest.get(Constants.NAME);
+		if (StringUtils.isBlank(name)) {
+			missingParams.add(Constants.NAME);
+		}
+		String refId = (String) designationRequest.get(Constants.REF_ID);
+		if (StringUtils.isBlank(refId)) {
+			missingParams.add(refId);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java]: suppose to add Constants.REF_ID

File: src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java [hunk 2]
```diff
@@ -432,4 +432,153 @@ private String validateSubSectorCreateReq(Map<String, Object> sectorRequest) {
...
+		}
+		String refType = (String) designationRequest.get(Constants.REF_TYPE);
+		if (StringUtils.isBlank(refType)) {
+			missingParams.add(refType);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java]: suppose to add Constants.REF_TYPE

File: src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java [hunk 3]
```diff
@@ -432,4 +432,153 @@ private String validateSubSectorCreateReq(Map<String, Object> sectorRequest) {
...
+			response.getParams().setStatus(Constants.FAILED);
+			response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		return response;
+	}
+
+	private Map<String, Object> createRequestBody(String id, String designation, List<String> refNodes) {
+		try {
+			Map<String, Object> reqBodyMap = new HashMap<>();
+			reqBodyMap.put(Constants.ID, id);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java]: why REF_ID and REF_TYPE are not used here ?
If those are not required then why those are mandatory?

File: src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java [hunk 4]
```diff
@@ -432,4 +432,153 @@ private String validateSubSectorCreateReq(Map<String, Object> sectorRequest) {
...
+					.append(extServerProperties.getOdcsCategoryName());
+			Map<String, Object> termResponse = outboundRequestHandlerServiceImpl.fetchResultUsingPost(strUrl.toString(),
+					createReq, null);
+			if (termResponse != null
+					&& Constants.OK.equalsIgnoreCase((String) termResponse.get(Constants.RESPONSE_CODE))) {
+				Map<String, Object> resultMap = (Map<String, Object>) termResponse.get(Constants.RESULT);
+				List<String> termIdentifier = (List<String>) resultMap.getOrDefault(Constants.NODE_ID, "");
+				log.info("Created Designation successfully with name: " + name);
+				log.info("termIdentifier : " + termIdentifier);
+				response = updateDesignation(refId, name, termIdentifier);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java]: what will happen if the updateDesignation fails ??

File: src/main/java/org/sunbird/catalog/controller/CatalogController.java [hunk 5]
```diff
@@ -46,4 +46,11 @@ public ResponseEntity<SBApiResponse> createSubSector(@RequestBody Map<String, Ob
 		SBApiResponse response = catalogService.createSubSector(request);
 		return new ResponseEntity<>(response, response.getResponseCode());
 	}
+
+	@PostMapping("/designation/create")
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/catalog/controller/CatalogController.java]: this implementation is in cb-ext --- just to reuse the framework capability -- but nothing is reused from existing code.... 

after creating term object - you are still calling pores service to update designation... 

instead - just update the endpoints (or add new one) - in pores service to create / update -- that should create a term object - if term creation fails it returns... if term is created then updates the db...

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java (hunk 3 above)
Comment (anilkumarkammalapalli) [reply src/main/java/org/sunbird/catalog/service/CatalogServiceImpl.java]: id is ref_id only here. we are passing from caller function.

**Pull request closed without merging**